sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,219 @@
+        clip_max[numeric_features] = self.n_bins_[numeric_features] - 1
+
+        self._clip_min = clip_min
+        self._clip_max = clip_max
+
+        same_min_max = np.where(self.ptp_ == 0)[0]
+        if len(same_min_max) > 0:
+            import warnings
+            warnings.warn("Fitted X contained continuous features at indices "
+                          "{}. All of these features will be discretized to "
+                          "the value 0."
+                          .format(", ".join(str(i) for i in same_min_max)))
+
+        return self
+
+    @staticmethod
+    def _check_n_bins(n_bins, n_features):
+        if isinstance(n_bins, numbers.Number):
+            if n_bins < 2:
+                raise ValueError("Discretizer received an invalid number "
+                                 "of bins. Received {0}, expected at least 2."
+                                 .format(n_bins))
+            return
+
+        n_bins = check_array(n_bins, dtype=int, ensure_2d=False)
+
+        if n_bins.ndim > 1 or n_bins.shape[0] != n_features:
+            raise ValueError("n_bins must be a scalar or array "
+                             "of shape (n_features_,).")
+
+        violating_indices = np.arange(n_features)[n_bins < 2]
+        if len(violating_indices) > 0:
+            indices = ", ".join(str(i) for i in violating_indices)
+            raise ValueError("Discretizer received an invalid number "
+                             "of bins at indices {}. Number of bins "
+                             "must be at least 2."
+                             .format(indices))
+
+    @staticmethod
+    def _check_categorical_features(cat_features, n_features):
+        if cat_features is None:
+            return
+
+        cat_features = check_array(cat_features, ensure_2d=False, dtype=int)
+        cat_features = column_or_1d(cat_features)
+
+        if len(set(cat_features)) != cat_features.shape[0]:
+            raise ValueError("Duplicate categorical column indices found.")
+
+        if np.all(cat_features >= 0) and np.all(cat_features < n_features):
+            return
+
+        raise ValueError("Invalid categorical feature index.")
+
+    def transform(self, X, y=None):
+        """Discretizes the data, ignoring any categorical features.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data containing features to be discretized.
+
+        Returns
+        -------
+        T : numeric array-like, shape (n_samples, n_features)
+            Categorical features will not be transformed.
+        """
+        check_is_fitted(self, ["min_", "ptp_", "_clip_min", "_clip_max"])
+        X = check_array(X, dtype=float, ensure_2d=False)
+
+        if X.ndim == 1:
+            raise ValueError("Reshape your data.")
+
+        n_features = self.n_bins_.shape[0]
+        if X.shape[1] != n_features:
+            raise ValueError("Transformed array does not have correct number "
```

Comment:
"transformed array" is a bit ambiguous. I think you can just say `X`. Or "Incorrect number of features"